bug: cannot build Rust UDF

On current main (ef8c50f5f56c04374cda3c8b8204afc10ee6a88f), Rust UDF cannot be built.

```sql
dev=> CREATE FUNCTION sum_array(integer[]) RETURNS integer LANGUAGE rust AS $$
    fn sum_array(xs: &[i32]) -> i32 {
        xs.iter().sum()
    }
$$;

ERROR:  Failed to run the query

Caused by these errors (recent errors listed first):
  1: failed to build rust function
  2: failed to build wasm (exit status: 101)
--- stdout

--- stderr
    Updating crates.io index
     Locking 3 packages to latest compatible versions
      Adding arrow-udf v0.3.0 (latest: v0.4.0)
      Adding arrow-udf-macros v0.3.0 (latest: v0.4.0)
      Adding lazy_static v1.5.0
   Compiling arrow-array v52.2.0
warning: the `wasm32-wasi` target is being renamed to `wasm32-wasip1` and the `wasm32-wasi` target will be removed from nightly in October 2024 and removed from stable Rust in January 2025

   Compiling arrow-select v52.2.0
   Compiling arrow-arith v52.2.0
   Compiling arrow-cast v52.2.0
warning: `arrow-array` (lib) generated 1 warning
   Compiling arrow-ipc v52.2.0
   Compiling arrow-udf v0.3.0
warning: `arrow-udf` (lib) generated 1 warning (1 duplicate)
warning: `arrow-select` (lib) generated 1 warning (1 duplicate)
warning: `arrow-ipc` (lib) generated 1 warning (1 duplicate)
warning: `arrow-arith` (lib) generated 1 warning (1 duplicate)
warning: `arrow-cast` (lib) generated 1 warning (1 duplicate)
   Compiling udf v0.1.0 (/private/var/folders/7r/p0w75cw53gj176x6_sxj7r8m0000gn/T/risingwave-rust-udf)
error[E0599]: no method named `as_primitive` found for struct `Arc<dyn arrow_array::array::Array>` in the current scope
   --> src/lib.rs:2:1
    |
2   | #[function("sum_array(int32[])->int32")]
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /Users/richard/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-array-52.2.0/src/cast.rs:750:8
    |
750 |     fn as_primitive<T: ArrowPrimitiveType>(&self) -> &PrimitiveArray<T> {
    |        ------------ the method is available for `Arc<dyn arrow_array::array::Array>` here
    |
    = help: items from traits can only be used if the trait is in scope
    = note: this error originates in the attribute macro `function` (in Nightly builds, run with -Z macro-backtrace for more info)
help: trait `AsArray` which provides `as_primitive` is implemented but not in scope; perhaps you want to import it
    |
1   + use arrow_udf::codegen::arrow_array::cast::AsArray;
    |
help: there is a method `as_primitive_opt` with a similar name
    |
2   | as_primitive_opt
    |

warning: unused import: `types`
 --> src/lib.rs:1:27
  |
1 | use arrow_udf::{function, types::*};
  |                           ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

For more information about this error, try `rustc --explain E0599`.
warning: `udf` (lib) generated 2 warnings (1 duplicate)
error: could not compile `udf` (lib) due to 1 previous error; 2 warnings emitted
```

This is possibly because of upgrading `arrow-udf` in some previous PR.

--- a/ci/scripts/build-other.sh
+++ b/ci/scripts/build-other.sh
@@ -8,7 +8,7 @@ source ci/scripts/common.sh
 echo "--- Build Rust UDF"
 cd e2e_test/udf/wasm
 rustup target add wasm32-wasi
-cargo build --release
+cargo build --release --target wasm32-wasi
 cd ../../..
 
 echo "--- Build Java packages"

--- a/e2e_test/udf/rust_udf.slt
+++ b/e2e_test/udf/rust_udf.slt
@@ -148,6 +148,23 @@ select series(3);
 statement ok
 drop function series;
 
+
+statement ok
+create function sum_array(integer[]) returns integer language rust as $$
+    fn sum_array(xs: &[i32]) -> i32 {
+        xs.iter().sum()
+    }
+$$;
+
+query I
+select sum_array(array[1, 2, 3]);
+----
+6
+
+statement ok
+drop function sum_array;
+
+
 # XXX: this test is disabled in CI because it prints "panicked at" which makes the test fail
 # statement ok
 # create function panic() returns int language rust as $$

--- a/e2e_test/udf/wasm/Cargo.toml
+++ b/e2e_test/udf/wasm/Cargo.toml
@@ -8,7 +8,7 @@ edition = "2021"
 crate-type = ["cdylib"]
 
 [dependencies]
-arrow-udf = "0.4"
+arrow-udf = "0.5"
 genawaiter = "0.99"
 rust_decimal = "1"
 serde_json = "1"

--- a/src/expr/impl/src/udf/wasm.rs
+++ b/src/expr/impl/src/udf/wasm.rs
@@ -92,7 +92,7 @@ fn create_rust(opts: CreateFunctionOptions<'_>) -> Result<CreateFunctionOutput> 
 
     let wasm_binary = std::thread::spawn(move || {
         let mut opts = arrow_udf_wasm::build::BuildOpts::default();
-        opts.arrow_udf_version = Some("0.3".to_string());
+        opts.arrow_udf_version = Some("0.5".to_string());
         opts.script = script;
         // use a fixed tempdir to reuse the build cache
         opts.tempdir = Some(std::env::temp_dir().join("risingwave-rust-udf"));

